src/DatePicker/DatePicker.js

```diff
@@ -231,7 +231,7 @@ class DatePicker extends Component {
   };
 
   isControlled() {
-    return this.props.hasOwnProperty('value');
+    return this.props.value !== null;
```

Comment:
I think that we should be using `undefined` to check for the controlled behavior.
Otherwise, how do you clear the date?

React is using the same approach for the `<input />` component. I get the following warning:
> Consider using the empty string to clear the component or `undefined` for uncontrolled components.